Update filesystem.cc

salpelter opened the pull request:
# Motivation
<!-- Briefly explain what the change is about and why it is desirable. -->
`filesystem.cc` is the only place where [`createSymlink()` with three arguments](https://github.com/NixOS/nix/blob/041486b11674e0480ad85f63d262c33a09276ef1/src/libutil/filesystem.cc#L89) is used, in the definition of replaceSymlink() with three parameters _is not used at all_.

# Context
<!-- Provide context. Reference open issues if available. -->
https://github.com/NixOS/nix/pull/7048 deals with timestamps, but there are occurrences that we can avoid handling altogether because they are dead code anyway.
<!-- Non-trivial change: Briefly outline the implementation strategy. -->

<!-- Invasive change: Discuss alternative designs or approaches you considered. -->

<!-- Large change: Provide instructions to reviewers how to read the diff. -->

# Checklist for maintainers

<!-- Contributors: please leave this as is -->

Maintainers: tick if completed or explain if not relevant

 - [ ] agreed on idea
 - [ ] agreed on implementation strategy
 - [ ] tests, as appropriate
   - functional tests - `tests/**.sh`
   - unit tests - `src/*/tests`
   - integration tests - `tests/nixos/*`
 - [ ] documentation in the manual
 - [ ] documentation in the internal API docs
 - [ ] code and comments are self-explanatory
 - [ ] commit message explains why the change was made
 - [ ] new feature or incompatible change: updated release notes

# Priorities

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

cole-h commented on src/libutil/filesystem.cc:
```diff
@@ -63,20 +63,10 @@ std::pair<AutoCloseFD, Path> createTempFile(const Path & prefix)
-    }
 }
 
 void replaceSymlink(const Path & target, const Path & link,
```
Seems like the `mtime` arg in `replaceSymlink` is now dead code too? Unless my eyes deceive me, it is no longer used anywhere.

fricklerhandwerk commented:
Please read up on how to write useful commit messages (and, by extension, pull request descriptions): https://cbea.ms/git-commit/

You can append commits while working on the PR and then squash them with a useful message when it's approved.

salpelter commented:
Hey @fricklerhandwerk, how do I create a follow-up change for this PR? If I commit changes like I did in the first place, will it appear here? If so, would the same name ("Remove dead code") be okay for the _commit message_? As for the _description_, if it's necessary, would it be okay to say that I removed some leftover dead code that I missed? And if i mention that, I should probably specify exactly what I removed, I guess. Sorry for so many little questions! but I think the faster I get rid of them the sooner I'll understand how to use GitHub properly. Thank you for the patience.
Also, you said that you linked the PR to the issue, should've I done it myself? If so, how do I do that?

fricklerhandwerk commented:
I recommend reading up on all of these questions yourself. It won't scale to hand-hold contributors through software development essentials. GitHub has great documentation, Git is widely used. Nix maintainers will assist with and share knowledge on Nix specifics.

salpelter commented:
Okay, understood!

Ericson2314 commented:
Thanks https://github.com/salpelter. Also yay, good to see the "good first issue" label work!
